src/commands/multi_block/monitor.rs

```diff
@@ -22,9 +22,117 @@ use polkadot_sdk::{
     pallet_election_provider_multi_block::unsigned::miner::MinerConfig,
     sp_npos_elections::ElectionScore,
 };
-use std::{collections::HashSet, sync::Arc};
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
 use tokio::sync::Mutex;
 
+/// Enum representing different tracking events for rounds
+#[derive(Debug, Clone, Copy)]
+enum RoundTrackingEvent {
+    Found,            // Found existing submission on-chain
+    Submitted,        // Successfully submitted a new solution
+    SubmittedPartial, // Submitted missing pages for partial solution
+    Cleared,          // Successfully cleared a discarded submission
+    FailedSubmission, // Failed to submit a solution
+}
+
+/// Enum representing different untracking events for rounds
+#[derive(Debug, Clone, Copy)]
+enum RoundUntrackingEvent {
+    Cleared, // Removed after clearing
+    Bailed,  // Removed after bailing
+    Failed,  // Removed after a failed submission
+}
+
+/// A dedicated type to manage submitted rounds tracking with clean APIs
+/// that hide the mutex implementation details
+#[derive(Clone)]
+struct RoundSubmission(Arc<Mutex<HashMap<u32, (u32, ElectionScore)>>>);
```

Comment:
nice I like the new the new type :)

I would personally move it `multi-block/types.rs` to make this file a bit shorter and easier to read but up to you :)